src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -73,14 +77,50 @@ public JobService(IamService samService,
 
         logger.info("Creating Stairway: maxStairwayThreads = " + appConfig.getMaxStairwayThreads());
         ExceptionSerializer serializer = new StairwayExceptionSerializer(objectMapper);
+        StairwayHook hooks = populateStairwayHooks();
         stairway = Stairway.newBuilder()
             .maxParallelFlights(appConfig.getMaxStairwayThreads())
             .exceptionSerializer(serializer)
             .applicationContext(applicationContext)
             .stairwayName(appConfig.getPodName())
+            .stairwayHook(hooks)
         .build();
     }
 
+    private StairwayHook populateStairwayHooks(){
```

Comment:
Is there a better place to put this chunk of code?

Reply:
I'd vote for making a separate derived class called `StairwayLoggingHooks` or some such, and then `new` it here. I am thinking that eventually, we may use some configuration setting to decide which of several hook implementations (or none) we want to create.

Reply:
This pattern makes a lot more sense, thank you!